### PR TITLE
Add future-compatible mongo Hook typing

### PR DESCRIPTION
**Pull request by potiuk**
We are migrating to new mongo library soon and it has typing support. This one adds propert typing that will prevent MyPy from failing once mongo gets upgraded.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by eladkal — APPROVED** [review]

**Comment by potiuk** [comment]
Looks like two more changes needed in `mongo_to_s3` :)

File: airflow/providers/mongo/hooks/mongo.py [hunk 1]
```diff
@@ -20,10 +20,12 @@
 
 from ssl import CERT_NONE
 from types import TracebackType
+from typing import Any, overload
 from urllib.parse import quote_plus, urlunsplit
 
 import pymongo
 from pymongo import MongoClient, ReplaceOne
+from typing_extensions import Literal
```
Comment (josh-fell) [inline airflow/providers/mongo/hooks/mongo.py]: Should `airflow.typing_compat` be used here?

Comment (potiuk) [reply airflow/providers/mongo/hooks/mongo.py]: of course. Silly me.

Comment (potiuk) [reply airflow/providers/mongo/hooks/mongo.py]: Fixed.

**Pull request merged**